docs/reference/experiment-config-reference.rst

```diff
@@ -13,9 +13,111 @@ The behavior of an experiment is configured via a YAML file. A configuration fil
 passed as a command-line argument when an experiment is created with the Determined CLI. For
 example:
 
-.. code::
-
-   det experiment create config-file.yaml model-directory
+.. tabs::
+
+   .. tab::
+
+      example
+
+      .. code::
+
+         det experiment create config-file.yaml model-directory
+
+   .. tab::
+
+      example with various configuration options
+
+      This sample contains comments that explain the purpose of sections and options without needing
+      to refer back to the documentation.
+
+      .. code:: yaml
+
+         # Sample YAML configuration file for an extensive experiment setup
+
+         metadata:
+         # Metadata for the experiment for easy identification and management
+            name: sample_experiment_name
+            description: "An example experiment configuration demonstrating various options."
+            tags:
+               - deep learning
+               - experiment
+
+         max_restarts: 3
+         # Maximum number of times the experiment is allowed to restart upon failure
+
+         environment:
+         # Environment settings specifying the runtime environment
+            image:
+               gpu: "determinedai/environments:cuda-11.3-pytorch-1.10-deepspeed-0.8.3-gpu-0.22.1"
```

Comment:
`environment_variables` would be a good addition here. The values underneath are of the form: `- "NAME=VALUE"` which is not at all obvious to everyone, even consulting the current docs.

Reply:
thanks!

Reply:
added this in with #9607